### PR TITLE
[FIX] Add new status

### PR DESCRIPTION
**Pull request by hurrinico**
I've create this PR in order to solve this Problem:
I create a Sale Order, validate and create an invoice from it, validate the invoice and pay it in the future (in a different period).
If i compute settle until today the invoice will be counted even if it's not already paid

### CODE REVIEW

**Comment by pedrobaeza** [comment]
I don't really understand the use case adding the move state, because I think paid already covers it.

**Comment by ivantodorovich** [comment]
I believe he doesn't want to settle invoices that are paid but payment was made in a date bigger than the "until date" field. At least not in that settlement.

The use case would be, for instance, settling only the invoices paid in the previous month (or before) but not including invoices that were paid in the current month (let's say.. in the very firsts days of the month, before the settlement wizard is ran).

Maybe this should be a FIX to "Payment Based" instead of adding another option, since it's actually what one would expect from this.

File: sale_commission/wizard/wizard_settle.py [hunk 1]
```diff
@@ -74,6 +74,14 @@ def action_settle(self):
                             agent_lines[pos].invoice.state != 'paid'):
                         pos += 1
                         continue
+                    if agent.commission.invoice_state == 'period':
```
Comment (pedrobaeza) [inline sale_commission/wizard/wizard_settle.py:77]: This should be 'move', not 'period'?

**Comment by pedrobaeza** [comment]
It's still very confusing...

File: sale_commission/wizard/wizard_settle.py (hunk 1 above)
Comment (ivantodorovich) [inline sale_commission/wizard/wizard_settle.py:77]: Indeed, that should be 'move'. 
Shouldn't it, @hurrinico ?

**Comment by ivantodorovich** [comment]
Lo escribo en español, porque sé que es también tu idioma principal, y creo que voy a poder explicarlo mejor.

Imaginate este escenario: 

La fecha es 10 de Febrero de 2016, y queremos liquidar las comisiones de Marcelo, nuestro agente, sobre facturas pagadas en forma mensual. 

El mes que corresponde liquidar es el mes de Enero, ya que Febrero se liquidará recién los primeros días de Marzo.

Pero resulta que a Juan, un cliente que paga siempre en término, tiene en su cuenta corriente una factura emitida el 25/01/2016, y cancelada el 05/02/2016.

No queremos que esa factura ingrese en la liquidación de Enero, ya que se pagó en Febrero. Debería ingresar recién en la próxima liquidación.

Este PR hace eso, asegurándose que no sólo la fecha de factura sea menor al 01/02/2016 (fecha que ingresaremos en el wizard) - y que dicha factura esté pagada -, sino también se asegura de que la fecha de su último pago (es decir, la fecha de cancelación) esté dentro del periodo a liquidar.

Esto, a mi parecer, es lo que uno esperaría al seleccionar 'Paid Based', es por eso que creo que los cambios sugeridos por @hurrinico , en lugar de crear una nueva opción de invoice_state, deberían ser incorporados dentro de 'paid'.

PD: Al menos así lo entiendo yo.. como siempre, puedo equivocarme :smile_cat:

**Comment by pedrobaeza** [comment]
Thanks for the explanation @ivantodorovich. I continue in English because we need to make each other understandable. Well, that can be, but let's see what @hurrinico says. Indeed currently I see now a potential problem where an invoice cannot be settled, because is not paid until the next period. @oihane, can you confirm this?

**Comment by hurrinico** [comment]
Hi @pedrobaeza sorry for delay but is a very busy period. 
I agree with you and @ivantodorovich i'll fix this PR leaving 2 states and applying my fix to the existing one.
I'll work on it soon (about friday i think).

**Comment by pedrobaeza** [comment]
Is this going to be continued?

**Comment by pedrobaeza** [comment]
Closing as no answer.

**Pull request closed without merging**